### ISSUE TITLE
Use XamlX for x:Name Resolution

### ISSUE THREAD

**Issue by worldbeater** [worldbeater/XamlNameReferenceGenerator#3]
The current source generator implementation is quite poor now. It doesn't handle namespaces properly. Currently, there is no support for scenarios e.g. when there are two controls with the same name but living in different namespaces etc. Ideally, we should incorporate XamlX and implement a type system based on C# source generators API, and use XamlX parser to parse and resolve the types. So, we should:

1. Implement a XamlX type system based on Roslyn source generation APIs https://github.com/kekekeks/XamlX/blob/master/src/XamlX/TypeSystem/TypeSystem.cs (see e.g. https://github.com/kekekeks/XamlX/blob/0028377ce7c7dc21f9fe71b45f62a95991b1ab58/src/XamlX/IL/SreTypeSystem.cs for sample type system implementation, for `x:Name`s we don't need to fully implement the type system)
2. Parse a `xaml` (`axaml`) file using XamlX and resolve `x:Name`s with fully qualified types.
 
Basically we just need to rewrite the [`NameReferenceXamlParser`](https://github.com/worldbeater/XamlNameReferenceGenerator/blob/main/XamlNameReferenceGenerator/NameReferenceXamlParser.cs#L10) to use `XamlX` instead of `XDocument`.

**Comment by worldbeater** [comment]
Guessing we just need to do the following:
```cs
var compiler = new AvaloniaXamlIlCompiler(new AvaloniaXamlIlCompilerConfiguration(_sreTypeSystem, asm,
                _sreMappings, _sreXmlns, AvaloniaXamlIlLanguage.CustomValueConverter,
                new XamlIlClrPropertyInfoEmitter(_sreTypeSystem.CreateTypeBuilder(clrPropertyBuilder)),
                new XamlIlPropertyInfoAccessorFactoryEmitter(_sreTypeSystem.CreateTypeBuilder(indexerClosureType))), 
                _sreEmitMappings,
                _sreContextType);

compiler.ParseAndCompile("xaml goes here", null, null, _sreTypeSystem.CreateTypeBuilder(tb), null);
```

**Comment by worldbeater** [comment]
Stumbled upon an issue with the inclusion of NuGet packages which is currently broken in C# source generator land, and requires a non-trivial dance with MSBuild properties https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.cookbook.md#use-functionality-from-nuget-packages Was unable to avoid that `System.IO.FileNotFoundException: Could not load file or assembly` thrown by the source generator.

Worth including [XamlX](https://github.com/kekekeks/XamlX) as a `git submodule` and via `<Compile Include="XamlX/**/*.cs"/>`.

**Comment by worldbeater** [comment]
As stated in https://github.com/worldbeater/XamlNameReferenceGenerator/pull/4, now we should incorporate `AvaloniaXamlIlCompiler`  https://github.com/AvaloniaUI/Avalonia/blob/cebd287a63fd4bb8d3a6a5c452ce7630272e5cca/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlCompiler.cs Probably worth referencing Avalonia main repo as a submodule in order to include transformers from there.
